### PR TITLE
feat: error message for custom revert error SF-764

### PR DESCRIPTION
**Pull request by ChaiSomsri96**
Since the custom error `TooManyActiveOrders()` is an error in the `LendingMarketUserLogic` library, it is not included in the ABI of the `LendingMarketController`. 
As a result, when a revert error occurs, viem cannot decode it. 
Therefore, to accurately display the `TooManyActiveOrders` error, the signature of `TooManyActiveOrders` needs to be defined on the frontend.
I created `errorMessages.ts` to include custom errors that we want to display in the future. The `TooManyActiveOrders` error can occur when calling the [`executeOrder`], [`depositAndExecuteOrder`], [`depositWithPermitAndExecuteOrder`], [`executePreOrder`], [`depositAndExecutesPreOrder`], and [`depositWithPermitAndExecutePreOrder`] functions of `LendingMarketController`. These functions are all integrated in components/organisms/PlaceOrder/PlaceOrder.tsx, which is why I have made modifications to PlaceOrder.tsx.

### CODE REVIEW

File: src/utils/errorMessages.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// errorMessages.ts
+
+// Error message mapping for different error signatures
+const errorMapping: Record<string, string> = {
+    // 'TooManyActiveOrders' corresponds to the error signature '0x28d7a641'
+    // error TooManyActiveOrders();
+    '0x28d7a641': 'TooManyActiveOrders',
+    // More mappings can be added here...
+};
```
Comment (biga816) [inline src/utils/errorMessages.ts:9]: With this way, we need to add all errors to be supported like this.
Is there any way to attach the library's ABI, like the following unit test using hardhat and ethers?

https://github.com/Secured-Finance/contracts/blob/4ba147ea4d7b617d61176930497495185d39c535/test/performance/order-book.test.ts#L101

File: src/components/organisms/PlaceOrder/PlaceOrder.tsx [hunk 2]
```diff
@@ -208,8 +209,14 @@ export const PlaceOrder = ({
                     dispatch({ type: 'next' });
                 }
             } catch (e) {
-                if (e instanceof Error) {
-                    setErrorMessage(e.message);
```
Comment (biga816) [inline src/components/organisms/PlaceOrder/PlaceOrder.tsx:212]: For unexpected errors, this logic is needed. In the current logic, only `ContractFunctionRevertedError` can set the error message.

**Review by biga816 — APPROVED** [review]

**Review by biga816 — APPROVED** [review]

File: src/hooks/useOrders/useOrders.ts [hunk 3]
```diff
@@ -48,6 +48,7 @@ export const useOrders = () => {
                 return tx;
             } catch (error) {
                 console.error(error);
+                throw error;
```
Comment (as472) [inline src/hooks/useOrders/useOrders.ts:51]: These errors can happen in placeOrder, placePreOrder, depositOrder, etc.
Do you know if the scope of the ticket is only to fix the placeOrder?

**Review by as472 — CHANGES_REQUESTED** [review]
Please check the review comments before merging.

File: src/hooks/useOrders/useOrders.ts (hunk 3 above)
Comment (biga816) [reply src/hooks/useOrders/useOrders.ts:51]: `TooManyActiveOrders` is just an example. We should make our app handle all custom errors that are handled in SDK.

**Review by as472 — CHANGES_REQUESTED** [review]
Please attach a UI screenshot displaying this error.

**Review by biga816 — COMMENTED** [review]
There are many duplicated code for error handling.
How about creating a common function for handling error in `src/hooks/useOrders/useOrders.ts` and calling it in each hook?

**Review by biga816 — APPROVED** [review]

**Review by as472 — APPROVED** [review]

**Pull request merged**